lib/hiera/config.rb

```diff
@@ -54,8 +54,7 @@ def validate!
         begin
           require "deep_merge"
         rescue LoadError
-          Hiera.error "Must have 'deep_merge' gem installed for the configured merge_behavior."
-          raise
+          raise "Must have 'deep_merge' gem installed for the configured merge_behavior."
```

Comment:
Hiera::InvalidConfigurationError is defined if we see it as an error in the configuration. Hiera::Error if we see this as a general (standard) error.